bootstraptest/test_yjit.rb

```diff
@@ -505,3 +505,76 @@ def itself
 
   [Iseq.call_itself, Cfunc.call_itself]
 }
+
+# attr_reader method
+assert_equal '[100, 299]', %q{
+class A
+  attr_reader :foo
+
+  def initialize
+    @foo = 100
+  end
+
+  # Make it extended
+  def fill!
+    @bar = @jojo = @as = @sdfsdf = @foo = 299
+  end
+end
+
+def bar(ins)
+  ins.foo
+end
+
+ins = A.new
+oth = A.new
+oth.fill!
+
+bar(ins)
+bar(oth)
+
+[bar(ins), bar(oth)]
```

Comment:
I do love that you added multiple tests for this 👍 

Very much a detail, but can we indent the tests since they are inside another structure (assert_equal)?